### PR TITLE
Log all accessible ports on remote mode dev

### LINKED ISSUE #1652: 🐛 BUG: default `0.0.0.0` wrangler dev address not routable in Windows
### What version of `Wrangler` are you using?

2.0.25

### What operating system are you using?

Windows 10

### Describe the Bug

When using `wrangler dev` or `wrangler pages dev`, you'll get output like this:

```
[pages:inf] Listening on 0.0.0.0:8788
[pages:inf] - http://0.0.0.0:8788
```

Previously, you could click on this and open up your browser without issue, as well as of course `b` opens this address too.

Unfortunately, at least within Windows, `0.0.0.0` isn't actually a routable address, and the browser throws a `ERR_ADDRESS_INVALID` error, and something like this: 

It seems like this address may be routable in some OSes, but not Windows. This seems to be due to the changes in #1605

### PR DESCRIPTION
**Pull request by Skye-31**
Closes #1652 
Uses borrowed logic from miniflare

### CODE REVIEW

File: packages/wrangler/src/proxy.ts [hunk 1]
```diff
@@ -484,3 +489,14 @@ export async function waitForPortToBeAvailable(
 		}
 	});
 }
+
+function getAccessibleHosts(): string[] {
+	const hosts: string[] = [];
+	Object.values(networkInterfaces()).forEach((net) => {
+		net?.forEach(({ family, address }) => {
+			// @ts-expect-error the `family` property is numeric as of Node.js 18.0.0
+			if (family === "IPv4" || family === 4) hosts.push(address);
```
Comment (WalshyDev) [inline packages/wrangler/src/proxy.ts:498]: How come we only do IPv4?

**Review by WalshyDev — COMMENTED** [review]
Just some questions, non blocking.

File: packages/wrangler/src/proxy.ts [hunk 2]
```diff
@@ -328,6 +329,10 @@ export function usePreviewServer({
 			.then(() => {
 				proxy.server.on("listening", () => {
 					logger.log(`⬣ Listening at ${localProtocol}://${ip}:${port}`);
```
Comment (WalshyDev) [inline packages/wrangler/src/proxy.ts:331]: Is there a reason we don't list them all even in the scenario of someone setting a specific one?

File: packages/wrangler/src/proxy.ts (hunk 1 above)
Comment (Skye-31) [reply packages/wrangler/src/proxy.ts:498]: I just copied miniflare's implementation to be consistent, happy to change, but then it should also be changed there

File: packages/wrangler/src/proxy.ts (hunk 2 above)
Comment (Skye-31) [reply packages/wrangler/src/proxy.ts:331]: If someone sets a specific one, it's very possible the others will not be being listened on, unlike 0.0.0.0 which listens on all addresses

**Comment by GregBrimble** [comment]
FYI, I've just put up a PR against Miniflare to fix a small bug related to the code you've copied: https://github.com/cloudflare/miniflare/pull/382

Not sure if it affects you here, but worth testing.

**Review by JacobMGEvans — CHANGES_REQUESTED** [review]
Lets make the comment from Greg a requirement before merging https://github.com/cloudflare/wrangler2/pull/1881#issuecomment-1251041038

**Comment by Skye-31** [comment]
Good catch, resolved

**Review by JacobMGEvans — APPROVED** [review]

**Comment by cameron-robey** [comment]
(Provided tests pass) I'm going to merge this in now

Doesn't look like we have proxy server tests? Maybe this is something we should address...

**Pull request merged**